src/server/readpool/config.rs

```diff
@@ -41,18 +51,40 @@ pub struct Config {
 impl Default for Config {
     fn default() -> Config {
         Config {
-            high_concurrency: DEFAULT_HIGH_CONCURRENCY,
-            normal_concurrency: DEFAULT_NORMAL_CONCURRENCY,
-            low_concurrency: DEFAULT_LOW_CONCURRENCY,
-            max_tasks_high: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_HIGH_CONCURRENCY,
-            max_tasks_normal: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_NORMAL_CONCURRENCY,
-            max_tasks_low: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_LOW_CONCURRENCY,
+            high_concurrency: DEFAULT_CONCURRENCY,
+            normal_concurrency: DEFAULT_CONCURRENCY,
+            low_concurrency: DEFAULT_CONCURRENCY,
+            max_tasks_high: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
+            max_tasks_normal: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
+            max_tasks_low: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
             stack_size: ReadableSize::mb(DEFAULT_STACK_SIZE_MB),
         }
     }
 }
 
 impl Config {
+    pub fn default_for_coprocessor() -> Config {
+        let cpu_num = sys_info::cpu_num().unwrap();
+        let concurrency = if cpu_num > 8 {
+            (f64::from(cpu_num) * 0.8) as usize
+        } else {
+            DEFAULT_COP_CONCURRENCY
+        };
+        Config {
+            high_concurrency: concurrency,
```

Comment:
Maybe we could define a function to create `Config`  with concurrency, so we needn't repeated on L53~60?